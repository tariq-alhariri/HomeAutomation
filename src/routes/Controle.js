global.self = global;
import React from 'react';
import {
    StyleSheet,
    View,
    Text,
    Image,
    TouchableOpacity,
    TouchableHighlight,
    TextInput,
    Button,
    KeyboardAvoidingView,
    Alert
} from 'react-native';
import SpeechAndroid from 'react-native-android-voice';
import { Icon} from 'react-native-elements'; 

const tts = require('react-native-android-speech')
export default class Controle extends React.Component {
        static navigationOptions={
            header:null,
        tabBarLabel:'Controle',
        tabBarIcon:({tintColor}) => (
        <Image source={require('./icon/control.png')}
         style={{width:24, height:24, tintColor:'white'}}>
        </Image>
       ) 
         
        
    }
        constructor(props) {
        super(props)
       this.state={
        init:this.getCureentUser(),
        wellcome:this.welcome(),
        name:"",
        turnon:"",
        connect:"",
        motion:"",
        text:"",
        temp:"",
        getTemp:this.temp(),
        autoDetect:this.autoDetect(),
        gasAutoDetect:this.gasAutoDetect()
       }
    }
    //motion auto detection
    autoDetect(){
        setInterval(async function(){
            try {

                 let response = await fetch('http://192.168.2.46:8000/motion');
                 let responseJson = await response.json();
                 if(responseJson=='y'){
                    Alert.alert("Warrning there is motion in your room")
                 }
           } catch(error) {
             console.error(error);
             }
        },2000);
    }
// Auto Gas Alarm 
gasAutoDetect(){
    setInterval(async function(){
        try {

             let response = await fetch('http://192.168.2.46:8000/gas');
             let responseJson = await response.json();
             if(responseJson=='g'){
                Alert.alert("Gas Danger")
                tts.speak({
                    text:'There is gas leaking in the kitchen, please do not play with electricity hurry up and close the gas buttle. In emergency cases call 911. ', 
                    pitch:1.5, 
                    forceStop : false , 
                    language : 'en', 
                    country : 'US' 
                }).then(isSpeaking=>{
                    //Success Callback
                    console.log(isSpeaking);
               }).catch(error=>{
                 //Errror Callback
                 console.log(error)
                  });
             }
       } catch(error) {
         console.error(error);
         }
    },2000);
}
// Read the temperature
     temp() {
        var x=this
        setInterval(async function(){
            try {

                 let response = await fetch('http://192.168.2.46:8000/temp');

                 let responseJson = await response.json();

                   // Alert.alert(typeof(responseJson));

                  x.setState({temp:responseJson});
                 //var x=JSON.parse(responseJson)
                // this.state.temp=responseJson

               //  res=JSON.parse(res)

           } catch(error) {

             console.error(error);

             }

        },10000);
        

    } 




    async speak(){
    try{
        //More Locales will be available upon release.
        var spokenText = await SpeechAndroid.startSpeech("Speak now", SpeechAndroid.ENGLISH);
        //setTimeout(function(){ alert("kokokoko"); }, 3000);
        //ToastAndroid.show(spokenText , ToastAndroid.LONG);
        this.state.text=spokenText
        var a = this.state.text.search("turn");
        var b = this.state.text.search("on");
        var c = this.state.text.search("off");  
        if(a !==-1 && b!==-1){
            tts.speak({
                text:'your fan turned on', 
                pitch:1.5, 
                forceStop : false , 
                language : 'en', 
                country : 'US' 
            }).then(isSpeaking=>{
    //Success Callback
                 console.log(isSpeaking);
            }).catch(error=>{
    //Errror Callback
                 console.log(error)
            });
          this.turnon();
        }
        else if(a !==-1 && c!==-1){
            tts.speak({
                text:'your fan turned off', 
                pitch:1.5, 
                forceStop : false , 
                language : 'en', 
                country : 'US' 
            }).then(isSpeaking=>{
    //Success Callback
                console.log(isSpeaking);
        }).catch(error=>{
    //Errror Callback
             console.log(error)
        });
          this.turnoff();
        }else{
            tts.speak({
                text:'i can not understand', 
                pitch:1.5, 
                forceStop : false , 
                language : 'en', 
                country : 'US' 
            }).then(isSpeaking=>{
    //Success Callback
                 console.log(isSpeaking);
            }).catch(error=>{
    //Errror Callback
                  console.log(error)
            });
          Alert.alert("Try again!")
        }


    }catch(error){
        switch(error){
            case SpeechAndroid.E_VOICE_CANCELLED:
                ToastAndroid.show("Voice Recognizer cancelled" , ToastAndroid.LONG);
                break;
            case SpeechAndroid.E_NO_MATCH:
                ToastAndroid.show("No match for what you said" , ToastAndroid.LONG);
                break;
            case SpeechAndroid.E_SERVER_ERROR:
                ToastAndroid.show("Google Server Error" , ToastAndroid.LONG);
                break;
        }
    }
    }
    //detect motion 
    welcome() {
        tts.speak({
            text:'welcome to home automation system , how can i help you',
            pitch:1.5, 
            forceStop : false , 
            language : 'en', 
            country : 'US' 
    }).then(isSpeaking=>{
    //Success Callback
    console.log(isSpeaking);
    }).catch(error=>{
    //Errror Callback
    console.log(error)
    });
    }
    async motion() {
        try {
                 let response = await fetch('http://192.168.2.46:8000/motion');
                 let responseJson = await response.json();
                 //responseJson=JSON.parse(responseJson)
                 
                 if(responseJson=='n'){
                    Alert.alert("No motion in room")
                    tts.speak({
                        text:'No motion in your room', 
                        pitch:1.5, 
                        forceStop : false , 
                        language : 'en', 
                        country : 'US' 
                    }).then(isSpeaking=>{
                        //Success Callback
                        console.log(isSpeaking);
                   }).catch(error=>{
                     //Errror Callback
                     console.log(error)
                      });
                 }else{
                    Alert.alert("there is motion in room");
                    tts.speak({
                        text:'there is motion in your room', 
                        pitch:1.5, 
                        forceStop : false , 
                        language : 'en', 
                        country : 'US' 
                    }).then(isSpeaking=>{
                        //Success Callback
                        console.log(isSpeaking);
                   }).catch(error=>{
                     //Errror Callback
                     console.log(error)
                      });
                 }
                 this.setState({motion:responseJson})
               //  res=JSON.parse(res)
           } catch(error) {
             console.error(error);
             }
    } 

    //get current user
    async getCureentUser() {
        try {
                 let response = await fetch('https://home99.herokuapp.com/user');
                 let responseJson = await response.json();
                 this.setState({name:responseJson.name})
           } catch(error) {
             console.error(error);
             }
     } 
async connect(){
        try {
                 let response = await fetch('http://192.168.2.46:8000/connect');
                 let responseJson = await response.json();
                if(responseJson=="already connected"){
                   Alert.alert("you already connected")
                }else{
                   Alert.alert("Connected")
                }
                 // if(responseJson){
                 //     Alert.alert("Connected")
                 // }
               
           } catch(error) {
             console.error(error);
             }
   }
   async turnon(){
        try {
                 let response = await fetch('http://192.168.2.46:8000/on');
                 let responseJson = await response.json();
               
           } catch(error) {
             console.error(error);
             }
   }
   async turnoff(){
        try {
                 let response = await fetch('http://192.168.2.46:8000/off');
                 let responseJson = await response.json();
                
           } catch(error) {
             console.error(error);
             }
   }

    render() {
        return (
            <View style={styles.container}>

            <View style={styles.logoContainer}>
                <Image
                    style={styles.logo}
                    source={require('./Smart.png')} />

            </View>
                <Text style={styles.header}>
                    Welcome: {this.state.name}
                </Text>
                <Text style={styles.header}>
                the temperature now : {this.state.temp}  Â°C
                </Text>
           
             <TouchableHighlight
            style={styles.buttonContainer}
             onPress={() => this.connect()}>
             <Text style={styles.buttonText}>Connect</Text>
             </TouchableHighlight>                
               

             <TouchableOpacity
             style={styles.buttonContainer}
             onPress={() => this.turnon()}>
             <Text style={styles.buttonText}>Turn on lights</Text>
             </TouchableOpacity> 
             
             <TouchableOpacity
             style={styles.buttonContainer}
             onPress={() => this.turnoff()}>
             <Text style={styles.buttonText}>Turn off lights</Text>
               </TouchableOpacity> 
                
             <TouchableOpacity
             style={styles.buttonContainer}
             onPress={() => this.motion()}>
             <Text style={styles.buttonText}>Detect motion in my room</Text>
             </TouchableOpacity> 
                
             <TouchableOpacity
             style={styles.buttonContainer}
             onPress={() => this.speak()}>
             <Text style={styles.buttonText}>Talk to your home</Text>
             </TouchableOpacity> 
            


            
        </View>
                

        )
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        backgroundColor: '#87CEFA',
        alignItems: 'center',
        paddingHorizontal: 20,
        paddingTop: 20,
        justifyContent: 'center'
    },
    
    logoContainer: {
        alignItems: 'center',
        flexGrow: 1,
        justifyContent: 'center'
    },

    logo: {
        width: 200,
        height: 200
    },
    header: {
        color: '#fff',
        marginTop: 5,
        textAlign: 'center',
        opacity: 0.8,
        marginBottom: 10,
        fontWeight: '700',
        fontSize :20
    },

    buttonContainer: {
        backgroundColor: '#94336A',
        paddingVertical: 15,
        marginBottom: 20,
        borderRadius:10,
        borderWidth: 1,
        borderColor: '#4F1335',
        height: 40,
        width: 200
    },
    buttonText: {
        textAlign: 'center',
        color: '#000000',
        fontWeight: '700'
    }
});