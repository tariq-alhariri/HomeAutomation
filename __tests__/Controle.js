import 'react-native';
import React from 'react';
import Controle from '../src/routes/Controle';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

test('Search snapshot test', () => {
  const component = renderer.create(<Controle />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot();
})
