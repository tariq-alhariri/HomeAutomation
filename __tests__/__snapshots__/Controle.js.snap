// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search snapshot test 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#87CEFA",
      "flex": 1,
      "justifyContent": "center",
      "paddingHorizontal": 20,
      "paddingTop": 20,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexGrow": 1,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "height": 200,
          "width": 200,
        }
      }
    />
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#fff",
        "fontSize": 20,
        "fontWeight": "700",
        "marginBottom": 10,
        "marginTop": 5,
        "opacity": 0.8,
        "textAlign": "center",
      }
    }
  >
    Welcome: 
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "#fff",
        "fontSize": 20,
        "fontWeight": "700",
        "marginBottom": 10,
        "marginTop": 5,
        "opacity": 0.8,
        "textAlign": "center",
      }
    }
  >
    the temperature now : 
    
      Â°C
  </Text>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "backgroundColor": "transparent",
        },
        Object {
          "backgroundColor": "#94336A",
          "borderColor": "#4F1335",
          "borderRadius": 10,
          "borderWidth": 1,
          "height": 40,
          "marginBottom": 20,
          "paddingVertical": 15,
          "width": 200,
        },
      ]
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000000",
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Connect
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#94336A",
        "borderColor": "#4F1335",
        "borderRadius": 10,
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 20,
        "opacity": 1,
        "paddingVertical": 15,
        "width": 200,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000000",
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Turn on lights
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#94336A",
        "borderColor": "#4F1335",
        "borderRadius": 10,
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 20,
        "opacity": 1,
        "paddingVertical": 15,
        "width": 200,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000000",
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Turn off lights
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#94336A",
        "borderColor": "#4F1335",
        "borderRadius": 10,
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 20,
        "opacity": 1,
        "paddingVertical": 15,
        "width": 200,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000000",
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Detect motion in my room
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "#94336A",
        "borderColor": "#4F1335",
        "borderRadius": 10,
        "borderWidth": 1,
        "height": 40,
        "marginBottom": 20,
        "opacity": 1,
        "paddingVertical": 15,
        "width": 200,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#000000",
          "fontWeight": "700",
          "textAlign": "center",
        }
      }
    >
      Talk to your home
    </Text>
  </View>
</View>
`;
